packages/action-tracker/src/index.ts

```diff
@@ -15,14 +15,30 @@ import {
 export const buildEmbraceMiddleware = () => {
   return (store: IMiddleware<IDispatch, any>) =>
     (next: IDispatch<IAnyAction>) =>
-    (action: IAnyAction) => {
+    async (action: IAnyAction) => {
       if (!next || typeof next !== "function") {
         console.warn(
           "[Embrace] The state managment middleware was not applied succesfully",
         );
         return 0;
       }
       const startTime = new Date().getTime();
+      const attributes = {
+        name: action.type.toString().toUpperCase(),
+        payload_size: zip(action.payload || 0).length.toString(),
+        "emb.type": "sys.rn_action",
+        outcome: "INCOMPLETE",
+      };
+      const spanId = await NativeModules.EmbraceManager.startSpan(
+        "emb-rn-action",
+        undefined,
+        startTime,
+      );
+      const attributePromises: Promise<boolean>[] = Object.entries(
```

Comment:
seems `attributePromises` is not needed anymore? Can the calls to `addSpanAttributeToSpan` just be made without adding the result to the list?